carbonj.service/src/main/java/com/demandware/carbonj/service/engine/MetricList.java

```diff
@@ -73,22 +77,36 @@ public boolean match(String name)
             return isBlackListed;
         }
 
-        List<Pattern> currentPatterns = patterns; // copy so we don't keep hitting the volatile barrier
-        for ( Pattern p : currentPatterns )
+        List<Pair<Pattern, Counter>> currentPatternsAndCounters = patternsAndCounters; // copy so we don't keep hitting the volatile barrier
+        for ( int i = 0; i < currentPatternsAndCounters.size(); i++ )
         {
-            if( ".*".equals( p.pattern() ) )
+            Pair<Pattern, Counter> p = currentPatternsAndCounters.get(i);
+
+            if( ".*".equals( p.getLeft().pattern() ) )
             {
+                patternsAndCounters.get(i).getRight().inc();
                 droppedMetrics.inc();
                 isBlackListed = true;
                 break;
             }
 
-            if ( p.matcher( name ).find() )
+            long startTime = 0;
+            if (log.isDebugEnabled()) {
+                startTime = System.nanoTime(); // Record start time
+            }
+            if ( p.getLeft().matcher( name ).find() )
             {
+                patternsAndCounters.get(i).getRight().inc();
```

Comment:
Rethink of it and wonder if this code change makes sense or not anymore due to the caching mechanism. We are not doing the pattern evaluation for every counter anymore. @diwanshu-crm @sholavanalli